components/compliance-service/api/tests/01_market_spec.rb

```diff
@@ -6,6 +6,15 @@
 
   def profiles ; Profiles::ProfilesService ; end
 
+  it "errors out when an invalid order field is specified" do
+    assert_grpc_error(/order field '1337' is invalid. Use either 'ASC' or 'DESC'/) do
+      GRPC profiles, :list, Profiles::Query.new(
+        sort: 'name',
+        order: 1337
+      )
+    end
+  end
```

Comment:
I fixed a small bug in this error message in #2320 and didn't add a test, and I also wanted to see what happened if I tried assigning an invalid value for this parameter to see if the error checking was even necessary. 

Maybe I'm missing something but it seems weird that an enum that only defines 0 or 1 as valid values allows assigning arbitrary values...

Reply:
that is super weird.
maybe it's b/c it's an integer?  still weird

Reply:
I found a technical explanation for some of what is happening, but it still seems like weird behavior to me:

> During deserialization, unrecognized enum values will be preserved in the message, though how this is represented when the message is deserialized is language-dependent. In languages that support open enum types with values outside the range of specified symbols, such as C++ and Go, the unknown enum value is simply stored as its underlying integer representation.

https://developers.google.com/protocol-buffers/docs/proto3#enum